src/store/modules/DownloadModule.ts

```diff
@@ -63,6 +64,13 @@ export const DownloadModule = {
             commit('removeDownload', download);
         },
 
+        retryDownloadById({state, dispatch}, downloadId: UUID) {
+            const download = state.allDownloads.find((d: DownloadProgress) => d.downloadId === downloadId);
+            if (download) {
```

Comment:
I'm not sure there's a scenario where the download object is not found, but currently this gives the user no indication that something went wrong and nothing's going to happen. Could we try to bake in error handling here?